docs/src/guide/hello_world.md

```diff
@@ -0,0 +1,290 @@
+# Hello world
+
+As the first step, we will build a simple application, starting with a classic Hello World, adding some state, and finally a remote API call. We will focus on the core, and return to the shell a little later, so unfortunately there won't be much to see until then.
+
+If you want to follow along, you should start by following the [Shared core and types](../getting_started/core.md), guide to set up the project.
+
+## Side-effects and capabilities
+
+I know we said we'd build a simple application, but we need to start by talking about side-effects a little. One of the key design choices in Crux is that the Core is free of side-effects (besides its internal state). Your application can never _perform_ anything that directly interacts with the environment around it - no network calls, no reading/writing files, and (somewhat obviously) not even updating the screen. Actually _doing_ all those things is the job of the Shell, the core can only _ask_ for them to be done.
+
+This makes the core portable between platforms, and, importantly, really easy to test. It also separates the intent, the "functional" requirements, from the implementation of the side-effects and the "non-functional" requirements (NFRs). For example, your application knows it wants to store data in a SQL database, but it doesn't need to know or care whether that database is local or remote. That decision can even change as the application evolves. You can read a lot more about how side-effects work in Crux in the chapter on [capabilities](./capabilities.md).
+
+To _ask_ the Shell for side effects, we will need to describe the possible side-effects the Core will request. And to help us create these requests (generated by the `Effect` derive macro), we can use capabilities - reusable bits of functionality which give us a nice API for performing side-effects. We'll look at them in a lot more detail later.
+
+```rust
+use serde::{Serialize, Deserialize};
+use crux_core::render::Render;
+use crux_macros::Effect;
+
+#[derive(Effect, Default)]
+struct Capabilities {
+    render: Render<Effect>
```

Comment:
Capabilities are generic over the event type not the effect type, so this isn't right. Since there's no event type below does this need to be:

```suggestion
    render: Render<()>
```

Although I'm also moderately sure that won't compile with the way the effect macro works now 😢

Reply:
Hmmm, this whole guide is out of date now, isn't it. 😬 Completely missed it, need to go and fix that... I'd rather have no docs than wrong docs